vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -937,6 +937,18 @@ void AdsImpl::OnServeAdNotification(
 
   const int rand = base::RandInt(0, eligible_ads.size() - 1);
   const CreativeAdNotificationInfo ad = eligible_ads.at(rand);
+
+  if (ad.priority == 0) {
+    FailedToServeAdNotification("Pacing ad delivery [0]");
+    return;
+  }
+
+  const int rand_priority = base::RandInt(0, ad.priority - 1);
+  if (rand_priority != 0) {
+    FailedToServeAdNotification("Pacing ad delivery");
+    return;
+  }
+
   ShowAdNotification(ad);
```

Comment:
not sure if I read this correctly but generally speaking - since this is applied after picking the eligible ad we basically reduce the chance of an ad opportunity being filled proportional to the priorities in the catalog. Do we have a sense of how much this will reduce ad impressions overall?

Reply:
Discussed in DM and good to go